### PR TITLE
PDF Report pipeline

### LINKED ISSUE #32: report: send summary reports on a schedule
### Task
- [x] create a schedule that generates a summary report on a daily basis
- [x] send the summary report to all the users that have subscribed

### User Story
As a developer, I need a scheduler so that this process can be automated and users can receive their summary reports on time.

### LINKED ISSUE #106: Connect Report pipeline as a lambda
### Task

Call ETL in pdf_report.py: 
- [x] Call extract performance of stations from RDS, 
- [x] Call transform into graphs 
- [x] Call load into S3 bucket and send as email to subscribers
- [x] Make it a lambda function 
- [x] Create docker file 

### User Story
As a developer, I want to create a lambda function to call ETL and generate a report, so that users can view the report.

### PR DESCRIPTION
**Pull request by FarihaChoudhury**
```pdf_report.py``` extracts yesterday's railway statistics from the RDS, transforms to generate statistical graphs, loads the report into an S3 bucket, and sends as an email attachment to subscribed users. 
Dockerfile for the pipeline lambda. 

closes #106, closes #32

### CODE REVIEW

File: pdf_report/pdf_report.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+from transform_pdf import transform_pdf
+from load_pdf import load_pdf
+
+REPORT_FILENAME = 'performance_report.pdf'
+
+
+def main(_event, _context):
+    """ Lambda handler to run the PDF report pipeline. """
+
+    logging.basicConfig(level=logging.INFO,
```
Comment (Zhi-704) [inline pdf_report/pdf_report.py]: This should be changed to `    logging.getLogger().setLevel(logging.INFO)` for logs to show on AWS

Comment (FarihaChoudhury) [reply pdf_report/pdf_report.py]: OH MY GOSH REALLY WOW

Comment (Zhi-704) [reply pdf_report/pdf_report.py]: YEAH I KNOW IT'S CRAZZZYYY

File: pdf_report/transform_pdf.py [hunk 2]
```diff
@@ -268,4 +288,5 @@ def transform_pdf() -> None:
 if __name__ == "__main__":
     logging.basicConfig(level=logging.INFO,
                         format="%(asctime)s - %(levelname)s - %(message)s")
+    load_dotenv()
     transform_pdf()
```
Comment (A-Faris) [inline pdf_report/transform_pdf.py]: Add file name here

File: pdf_report/extract_pdf.py [hunk 3]
```diff
@@ -28,15 +27,20 @@ def get_cursor(conn: connection) -> cursor:
...
-    return data
+    try:
+        with get_cursor(conn) as cur:
+            cur.execute(query)
+            data = cur.fetchall()
+        return data
+
+    except:
+        logging.error("Error occurred when executing query: %s", query)
+        return
```
Comment (A-Faris) [inline pdf_report/extract_pdf.py]:
```suggestion
        return None
```

File: pdf_report/extract_pdf.py [hunk 4]
```diff
@@ -28,15 +27,20 @@ def get_cursor(conn: connection) -> cursor:
...
-    logging.info("Database query successful: %s", query)
-    return data
+    try:
+        with get_cursor(conn) as cur:
+            cur.execute(query)
+            data = cur.fetchall()
+        return data
+
+    except:
+        logging.error("Error occurred when executing query: %s", query)
```
Comment (A-Faris) [inline pdf_report/extract_pdf.py]:
```suggestion
    except Exception as e: # pylint: skip the error here
        logging.error("Error occurred when executing query: %s", query)
        logging.error("Error: %s", e)
```

File: pdf_report/load_pdf.py [hunk 5]
```diff
@@ -16,8 +14,7 @@
 from psycopg2.extensions import connection, cursor
 from psycopg2.extras import RealDictCursor
```
Comment (A-Faris) [inline pdf_report/load_pdf.py]: Delete these

File: pdf_report/load_pdf.py [hunk 6]
```diff
@@ -145,28 +117,27 @@ def send_email(ses_client: client, sender: str, subscribers: list[str], msg: MIM
...
+    email_performance_report(report_filename)
+    upload_pdf_to_s3(report_filename)
 
 
 if __name__ == "__main__":
     logging.basicConfig(level=logging.INFO,
                         format="%(asctime)s - %(levelname)s - %(message)s")
-
-    send_pdf_to_email_and_s3()
+    load_pdf()
```
Comment (A-Faris) [inline pdf_report/load_pdf.py]: Add file name

File: pdf_report/load_pdf.py [hunk 7]
```diff
@@ -56,24 +46,6 @@ def upload_pdf_to_s3() -> None:
             "Error occurred when connecting and uploading to S3 bucket.")
```
Comment (A-Faris) [inline pdf_report/load_pdf.py:45]: Skip error `# pylint: skip broad exception`

File: pdf_report/test_extract_pdf.py
Comment (A-Faris) [inline pdf_report/test_extract_pdf.py:1]: Nice test

**Review by A-Faris — DISMISSED** [review]
Solid foundation just pylint

**Review by Zhi-704 — DISMISSED** [review]
Looks good!

File: pdf_report/pdf_report.py
Comment (A-Faris) [inline pdf_report/pdf_report.py:1]: Docstring missing

File: pdf_report/pdf_report.py [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        logging.info("Report PDF pipeline has started")
+
+        transform_pdf(REPORT_FILENAME)
+        logging.info(
+            "Extracting & Transforming to create a PDF Summary Report complete.")
+
+        load_pdf(REPORT_FILENAME)
+        logging.info("Loading to S3 and sending emails complete.")
+
+    except Exception as e:
```
Comment (A-Faris) [inline pdf_report/pdf_report.py]: Add pylint skip here as well

**Review by A-Faris — DISMISSED** [review]
Looks great but please fix pylint and look at all except statements

**Review by Zhi-704 — APPROVED** [review]
Freed from suffering

**Pull request merged**